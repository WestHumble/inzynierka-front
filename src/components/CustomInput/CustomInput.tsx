import { View, TextInput, StyleSheet, Platform } from "react-native";
import React from "react";

const CustomInput = ({
  value,
  setValue,
  placeholder,
  secureTextEntry,
  additionalStyle,
  inputType = "text", // DomyÅ›lnie ustawiamy na "text"
}) => {
  const InputComponent = inputType === "textArea" ? TextInput : TextInput;

  return (
    <View style={[styles.container, additionalStyle]}>
      <InputComponent
        value={value}
        onChangeText={setValue}
        placeholder={placeholder}
        style={[
          styles.input,
          additionalStyle,
          inputType === "textArea" && styles.textArea,
        ]}
        placeholderTextColor="#f9f9f9"
        multiline={inputType === "textArea"}
        secureTextEntry={secureTextEntry}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderWidth: 1,
    borderRadius: 20,
    borderColor: "#131417",
    paddingHorizontal: 10,
    marginVertical: 5,
    backgroundColor: "#131417",
    width: "100%",
    color: "#fff",
  },
  input: {
    height: 60,
    ...Platform.select({
      ios: {
        // Dodatkowe style dla iOS
      },
      android: {
        // Dodatkowe style dla Androida
      },
    }),
  },
  textArea: {
    height: 120,
  },
});

export default CustomInput;
