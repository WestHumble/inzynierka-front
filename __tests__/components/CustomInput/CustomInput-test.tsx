import React, {useState} from 'react';
import renderer from 'react-test-renderer';
import { expect, test } from '@jest/globals';
import CustomInput from "../../../src/components/CustomInput";

test('renders correctly', () => {
    const tree = renderer.create(<CustomInput
        placeholder="Name"
        value={"value"}
        setValue={()=>{}}
    />).toJSON();
    expect(tree).toMatchSnapshot();
});                                                                                                  ``