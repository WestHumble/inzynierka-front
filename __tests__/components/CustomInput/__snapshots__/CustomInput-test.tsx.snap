// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#131417",
        "borderColor": "#131417",
        "borderRadius": 20,
        "borderWidth": 1,
        "color": "#fff",
        "marginVertical": 5,
        "paddingHorizontal": 10,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <TextInput
    editable={true}
    multiline={false}
    onChangeText={[Function]}
    placeholder="Name"
    placeholderTextColor="#f9f9f9"
    style={
      [
        {
          "height": 60,
        },
        undefined,
        false,
      ]
    }
    value="value"
  />
</View>
`;
